source/common/network/listen_socket_impl.cc

```diff
@@ -43,14 +42,12 @@ TcpListenSocket::TcpListenSocket(Address::InstanceConstSharedPtr address, bool b
   }
 }
 
-TcpListenSocket::TcpListenSocket(int fd, Address::InstanceConstSharedPtr address) {
-  fd_ = fd;
-  local_address_ = address;
-}
+TcpListenSocket::TcpListenSocket(int fd, const Address::InstanceConstSharedPtr& address)
+    : ListenSocketImpl(fd, address) {}
 
-UdsListenSocket::UdsListenSocket(const std::string& uds_path) {
+UdsListenSocket::UdsListenSocket(const std::string& uds_path)
+    : ListenSocketImpl(-1, std::make_shared<Address::PipeInstance>(uds_path)) {
   remove(uds_path.c_str());
-  local_address_.reset(new Address::PipeInstance(uds_path));
```

Comment:
In the old code this was done after `remove` and now it's done before. Does that matter? Or does `remove` only need to be done before bind potentially?